vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -2059,6 +2073,322 @@ private void createNotBufferedDynamicEditor() {
+                personService.fetch(0, 50));
+        List<Person> availablePersons = new ArrayList<Person>(
+                personService.fetch(0, 50));
+
+        grid.setItems(persons);
+
+        grid.addDragStartListener(
+                event -> draggedItems = event.getDraggedItems());
+        grid.setColumns("firstName", "lastName", "phoneNumber");
+        grid.setRowsDraggable(true);
+        grid.setDropMode(DropMode.BETWEEN);
+        grid.setSelectionMode(Grid.SelectionMode.MULTI);
+        grid.addDragStartListener(
+                event -> draggedItems = event.getDraggedItems());
+
+        grid.setDragDataGenerator("text",
+                person -> String.valueOf(person.getId()));
+
+        grid.addDropListener(event -> {
+            String text = event.getDataTransferText();
+            // The drag data tokens are separated by line breaks
+            Arrays.asList(text.split("\n")).forEach(idString -> {
+                Optional<Person> matchOptional = availablePersons
+                        .stream().filter(person -> String
+                                .valueOf(person.getId()).equals(idString))
+                        .findFirst();
+
+                if (matchOptional.isPresent()) {
+                    Person match = matchOptional.get();
+                    Person clone = new Person(match.getId(),
+                            match.getfirstName(), match.getLastName(),
+                            match.getAge(), match.getAddress(),
+                            match.getPhoneNumber());
+
+                    int dropIndex = persons
+                            .indexOf(event.getDropTargetRow().get())
+                            + (event.getDropLocation() == DropLocation.BELOW ? 1
+                                    : 0);
+                    persons.add(dropIndex, clone);
+                }
+            });
+            grid.getDataProvider().refreshAll();
+        });
+
+        grid.addDragEndListener(event -> {
+            // Additional validation logic might be needed to verify
+            // the drop was successful (on another browser window)
+            // The demo just removes the items regardless of whether the drop
+            // was legal.
+            persons.removeAll(draggedItems);
+            grid.getDataProvider().refreshAll();
+        });
+
+        // end-source-example
+
+        addCard("Drag and Drop", "Custom Drag Data", grid);
+    }
+
+    private void createDragDropFilters() {
+        // begin-source-example
+        // source-example-heading: Drag and Drop Filters
+        PersonService personService = new PersonService();
+
+        TreeGrid<Person> grid = new TreeGrid<>();
+        TreeData<Person> td = new TreeData<>();
+
+        // Disallow dragging supervisors
+        grid.setDragFilter(person -> td.getParent(person) != null);
+
+        grid.setDropFilter(person -> {
+            return td.getRootItems().contains(person) // Only support dropping
```

Comment:
Autoformatting makes these comments ugly. Move them to separate lines.

In case you think the auto-formatting should be turned off for a code-example, you can surround it with comments:
```
//@formatter:off
code here
//@formatter:on
```
This makes the IDE ignore these lines when running the formatter.

Maybe not necessary in this case, but good to know for the future at least.

Reply:
Done.